get_history.py

```diff
@@ -0,0 +1,63 @@
+from configuration import Configuration
+from database import Database
+from payment import Payment
+
+from pydantic import BaseModel, ValidationError, field_validator
+from datetime import datetime as date
+import json
+import sys
+
+
+class Record(BaseModel):
+    payer: str
+    amount: str
+    wallet: str
+    note: str
+    datetime: str
+
+    @field_validator("datetime")
+    def validate_timestamp(cls, value):
+        try:
+            timestamp = date.fromisoformat(value)
+            return date.strftime(timestamp, "%Y-%m-%d %H:%M:%S")
+        except ValueError:
+            raise ValidationError("Invalid timestamp format")
+
+
+def past_records(input_json: str) -> None:
+    config = Configuration()
+    database = Database(config)
+
+    try:
+        past_records_json = json.loads(input_json)
+    except json.JSONDecodeError as e:
+        raise ValueError("The input data is not in json format.")
+
+    for record in past_records_json.get("payments", []):
+        try:
+            record = Record(**record)
+        except ValidationError as e:
+            print(f"Invalid dictionary {record}: {e}")
+            continue
```

Comment:
I see that we first check if a record is valid, then we write it to the table. The problem is, for example if we have 20 records, and if record number 5 is invalid, we have already written 4 records to the table.
I think if there is any invalid records, we should throw an error and not write anything.

Also I think we should pass the usernames and wallets to this script, so that we can be sure that the record's payer, payee and wallet are as expected.